### DISCUSSION TITLE
Unable to open GPIO direction interface for pin

### DISCUSSION THREAD

**Discussion by zhouhuabin** [Pi4J/pi4j#79] [General]
I was using pi4j 1.3 to test Gpio examples on raspberry 4b, but i can not export the pin  3,7,28 with RaspiPinNumberingScheme.DEFAULT_PIN_NUMBERING,other pin number is ok.  and i run  command in /sys/classes/gpio echo 7 > export  is also ok.  i can not understand why this happen!
output:
platform:RASPBERRYPI
<--Pi4J--> GPIO test program
version:1.3
Exception in thread "main" java.lang.RuntimeException: Unable to open GPIO direction interface for pin [7]: No such file or directory
	at com.pi4j.wiringpi.GpioUtil.export(Native Method)
	at WiringPiGpioExample.main(WiringPiGpioExample.java:83)

main code is:
 URLClassLoader cl = (URLClassLoader) (com.pi4j.platform.PlatformManager.class.getClassLoader());
		URL url = cl.findResource("META-INF/MANIFEST.MF");
		Manifest manifest = new Manifest(url.openStream());
		Attributes attributes = manifest.getMainAttributes();
		String version = attributes.getValue("Bundle-Version");//Implementation-
		System.out.println("version:"+version);
        // setup wiringPi
        if (Gpio.wiringPiSetup() == -1) {
            System.out.println(" ==>> GPIO SETUP FAILED");
            return;
        }
        GpioFactory.setDefaultProvider(new RaspiGpioProvider(RaspiPinNumberingScheme.DEFAULT_PIN_NUMBERING));
        // set GPIO 4 as the input trigger
        GpioUtil.export(7, GpioUtil.DIRECTION_IN);
        GpioUtil.setEdgeDetection(7, GpioUtil.EDGE_BOTH);
        Gpio.pinMode (7, Gpio.INPUT) ;
        Gpio.pullUpDnControl(7, Gpio.PUD_DOWN);

**Comment by jmspaggi** [comment]
Him I think I'm facing something similar. Did you resolve your issue?

**Comment by FDelporte** [comment]
Are you using WiringPi or BCM numbers?
Seel also my comment on https://github.com/Pi4J/pi4j-v2/discussions/107

**Reply by jmspaggi** [reply]
Will reply there.

**Comment by zhouhuabin** [comment]
Him,
    I resolved my issue , and avoid using  gpio.provisionDigitalInputPin method :
gpio.provisionDigitalInputPin(RaspiBcmPin.GPIO_22, "input",PinPullResistance.PULL_UP);
  I just use Gpio.pinMode(bcm_TiltPin, Gpio.INPUT);

  then read value like this:
       int value = Gpio.digitalRead(bcm_TiltPin);
  not use :
       int value = aihome_tiltPin.getState().getValue();

Good luck!



***@***.***
 
From: jmspaggi
Date: 2021-06-04 01:06
To: Pi4J/pi4j-v2
CC: 周华彬; Author
Subject: Re: [Pi4J/pi4j-v2] Unable to open GPIO direction interface for pin (#79)
Him I think I'm facing something similar. Did you resolve your issue?
